docs/src/pages/system/basics/basics.md

```diff
@@ -139,94 +139,21 @@ If you are already using `@material-ui/core`, we encourage you to start with the
 **All** the properties are responsive, we support 3 different APIs.
 It uses this default, but customizable, breakpoints theme structure:
 
-```js
-const values = {
-  xs: 0,
-  sm: 600,
-  md: 960,
-  lg: 1280,
-  xl: 1920,
-};
-
-const theme = {
-  breakpoints: {
-    keys: ['xs', 'sm', 'md', 'lg', 'xl'],
-    up: key => `@media (min-width:${values[key]}px)`,
-  },
-};
-```
+{{"demo": "pages/system/basics/ResponsiveValues.js", "defaultCodeOpen": true}}
```

Comment:
Do we need a demo here? Shouldn't we move the TypeScript code as a test case instead?

Reply:
Do we have this documented somewhere else?

Reply:
To be clear, we need to document the API as well as to make sure that it works with TypeScript. I'm not sure it's two concerns we should merge. My suggestion is that we keep the markdown but we move the live version to a TypeScript test file.

Reply:
There doesn't appear to by anything in place for a typescript test suite; it was my understanding that these typescript demos are the tests. Should I just restore the js markdown (so we don't have the "demo" in the docs) but leave the `ResponsiveValues.tsx` file so it can be "tested"?

If I do that, and intentionally screw up something with the ResponsiveValues "demo", the `yarn typescript` command fails like it should, and we don't have to have the export because it's not being used as a demo.

Once I get an answer on this, I'll commit & push everything that was requested in this round of review.

Reply:
We do have dedicated TypeScript tests. Demos are not the tests but they help with the coverage. I have no problem with a live demo, as long as it's optimized for learning, not testing.

Reply:
Oh well in that case I'll probably leave it as-is. I originally just copy/pasted from the markdown into a typescript file and added types for reference. 

The only reason it has a default export is because the demo component appears to require an export. It doesn't render anything, it just shows the code.

Other than that, the docs for this section are unchanged, so we get the original code with a typescript variant + coverage from the tsx file.

Reply:
If it just show the code, it should be in the markdown, not in the runtime.

Reply:
Okay, I'll move it back to the markdown. Can you help me find the typescript tests? I'm struggling to find them at the moment.

Reply:
I moved it back into the markdown and removed the demo file for this. I don't think it needs extra testing, as the theme stuff already has extensive tests in `packages/material-ui-styles/test/*.spec.tsx`.